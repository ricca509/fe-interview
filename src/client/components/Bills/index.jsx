import React from 'react';
import PropTypes from 'prop-types';
import Bill from '../Bill';

const Bills = ({ bills = [], onActionClicked }) => {
  return (
    <ul>
      {bills.map(bill => (
        <Bill
          {...bill}
          key={bill.id}
          onActionClicked={onActionClicked}
        />
      ))}
    </ul>
  );
};

Bills.propTypes = {
  bills: PropTypes.array,
  onActionClicked: PropTypes.func
};

export default Bills;