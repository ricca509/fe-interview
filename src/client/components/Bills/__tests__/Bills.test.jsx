import React from 'react';
import Bills from '../';
import { shallow } from 'enzyme';

describe('Bill component', ()=> {
  const props = {
    bills: [
      {
        "name": "Vodafone",
        "isBill": false,
        "transactions": [
          {
            "amount": 12.34,
            "date": "2017-01-13"
          }
        ],
        "id": "5a5caa1efe33900100fd8ed5"
      },
      {
        "name": "Sky TV",
        "isBill": true,
        "transactions": [
          {
            "amount": 82.17,
            "date": "2017-01-01"
          }
        ],
        "id": "5a5caa8efe33900100fd8ed6"
      },
    ]
  };

  it('should render a list of `Bill`', () => {
    const wrapper = shallow(<Bills {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass the right callback to every `Bill`', () => {
    const onActionClickedMock = jest.fn();
    const ownProps = {
      ...props,
      onActionClicked: onActionClickedMock
    };
    const wrapper = shallow(<Bills {...ownProps} />);

    expect(wrapper.find('Bill').at(0).prop('onActionClicked')).toBe(onActionClickedMock);
    expect(wrapper.find('Bill').at(1).prop('onActionClicked')).toBe(onActionClickedMock);
  });
});