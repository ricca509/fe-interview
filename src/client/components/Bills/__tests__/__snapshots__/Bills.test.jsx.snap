// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill component should render a list of \`Bill\` 1`] = `
<ul>
  <Bill
    id="5a5caa1efe33900100fd8ed5"
    isBill={false}
    key="5a5caa1efe33900100fd8ed5"
    name="Vodafone"
    transactions={
      Array [
        Object {
          "amount": 12.34,
          "date": "2017-01-13",
        },
      ]
    }
  />
  <Bill
    id="5a5caa8efe33900100fd8ed6"
    isBill={true}
    key="5a5caa8efe33900100fd8ed6"
    name="Sky TV"
    transactions={
      Array [
        Object {
          "amount": 82.17,
          "date": "2017-01-01",
        },
      ]
    }
  />
</ul>
`;
